.github/workflows/build-test.yml

```diff
@@ -709,300 +650,32 @@ jobs:
-    # Cargo-make boilerplate
-    - name: Get cargo-make version
-      id: cargo-make-version
-      run: |
-        echo "hash=$(cargo search cargo-make | grep '^cargo-make =' | md5sum)" >> $GITHUB_OUTPUT
-      shell: bash
-    - name: Attempt to load cached cargo-make
-      uses: actions/cache@v3
-      id: cargo-make-cache
-      with:
-        path: |
-          ~/.cargo/bin/cargo-make
-          ~/.cargo/bin/cargo-make.exe
-        key: ${{ runner.os }}-make-${{ steps.cargo-make-version.outputs.hash }}
-    - name: Install cargo-make
-      if: steps.cargo-make-cache.outputs.cache-hit != 'true'
-      run: cargo +stable install cargo-make
-
-
-    # Toolchain boilerplate
-    - name: Potentially override rust version with nightly
-      run: cargo make set-nightly-version-for-ci
-    - name: Show the selected Rust toolchain
-      run: rustup show
-
-    - name: Load the default Rust toolchain via the rust-toolchain file.
-      run: rustup show
-
-    - name: Create output dir
-      run: mkdir -p ./benchmarks/perf/${{ matrix.component }}
-
-    ## Actual job
-    ##
-
-    # Benchmarking & dashboards job > Run benchmark.
-
-    - name: Run benchmark
-      run: |
-        pushd $PWD && cd ${{ matrix.component }};
-        export REL_OUTPUT_PATH="`dirs +1`/benchmarks/perf/${{ matrix.component }}";
-        mkdir -p $REL_OUTPUT_PATH;
-        export OUTPUT_PATH_CMD="ls -d $REL_OUTPUT_PATH";
-        export OUTPUT_PATH=$(echo $OUTPUT_PATH_CMD | sh);
-        cargo bench --features bench -- --output-format bencher | tee $OUTPUT_PATH/output.txt;
-        popd
-
-    # In the following step(s) regarding converting benchmark output to dashboards, the branch in `gh-pages-branch` needs to exist.
-    # If it doesn't already exist, it should be created by someone with push permissions, like so:
-    #   # Create a local branch
-    #   $ git checkout --orphan <newbranch>
-    #   $ git commit --allow-empty -m "root commit"
-    #   # Push it to create a remote branch
-    #   $ git push origin <newbranch>:<newbranch>
-
-    # Benchmarking & dashboards job > (unmerged PR only) Convert benchmark output into dashboard HTML in a commit of a branch of the local repo.
-
-    - name: Store benchmark result & create dashboard (unmerged PR only)
-      if: github.event_name == 'pull_request'
-      uses: rhysd/github-action-benchmark@v1.15.0
-      with:
-        name: Rust Benchmark
-        tool: 'cargo'
-        output-file-path: ./benchmarks/perf/${{ matrix.component }}/output.txt
-        benchmark-data-dir-path: ./benchmarks/perf/${{ matrix.component }}
-        # Show alert with commit comment on detecting possible performance regression
-        alert-threshold: '200%'  # If for nothing else, enabling the possibility of alerts with meaningful thresholds requires this job to be done per-component
-        fail-on-alert: true
-        gh-pages-branch: unmerged-pr-bench-data  # Requires one-time-only creation of this branch on remote repo.
```

Comment:
Note: This branch looks like it was deleted on the upstream repo, which is the reason why open PRs are failing and some past merges to main since Jan 31 failed. I went ahead restored the branch to allow other open PRs to get past that problem. This PR will obviate the need for the branch, so merging this PR should have come first.